### ISSUE TITLE
YN-300 III

### ISSUE THREAD

**Issue by yamisama** [kenkeiter/lantern#2]
I just tried the commands on a Yongnuo YN-300 III (I have the daylight-balanced single-color version, but a bi-color version also exists). 

I couldn't use the code here on iOS since talking to Bluetooth LE from Pythonista requires using their _cb_ module (a wrapper for CoreBluetooth, which unfortunately makes for really messy code due to its excessive use of the delegate programming paradigm). Would be cool if there was a cross-platform way to do Bluetooth LE. But I got it to work.

It identifies as "YONGNUO LED", and the write characteristic UUID is exactly the same as for the YN-360.

It turns out the control sequence for the 5500K LED intensity also works fine, and by extension, it's probably safe to assume that the bi-color version would support the 3200K octet. 

The YN-300 III doesn't have any RGB LEDs, so RGB mode is obviously not available. The power off command sequence also did not seem to have any discernible effect for me. Specifically, it didn't turn the light off.

I hooked the set intensity send function up to a UI slider, and latency seems much better with much more accurate control than with the official Yongnuo iOS app with its strange touch strip interface.

**Comment by kenkeiter** [comment]
@yamisama Would love a PR to add that compatibility, if you have time to submit one! 

I can probably try to RE the "YN-300 III" (I have a couple of those) and get support for it in there. I'll  need to find _some_ attribute to differentiate it from the YN-300, though... If I remember correctly, the YN-300 just identifies as "YONGNUO"? Anyway, I'll figure it out. Thanks for taking the time to write in 👍

**Comment by yamisama** [comment]
Thanks for your reply!

I don't currently have a desktop computer or laptop with Bluetooth LE support, so I don't actually have a way to run your code, hence a pull request would probably be difficult. The test I have is for Pythonista on iOS only, and it's pretty much the messiest test code I have written in my life (I was trying to figure out how Bluetooth LE works, how this CoreBluetooth wrapper with its delegate system is supposed to be used, and Pythonista's GUI module at the same time). But I'd of course be happy to share it if there is interest.

I think your code would probably already work with the YN-300 III without and major adjustments since the control sequence and UUID of the write characteristic are essentially the same. The RGB, off, and the setting for warm LEDs on the non-bi-color version would probably be ignored by the device, but other than that I think it should work right out of the box.

My YN-300 III shows up with "YONGNUO LED" as a device name. I don't know if there is actually a way to differentiate between the different models. The "BLE Scanner" iOS app, which I found really handy, does show a device UUID for every peripheral. But I don't know enough about how Bluetooth works – that field might just be a serial number or an iOS specific way to obfuscate the Bluetooth MAC address for privacy reasons or something like that. 

I wonder if the official Yongnuo Android and iOS control apps are able to differentiate between models.

There must be a Bluetooth-based "off" command for the YN-300 III, though, since the supplied remote has an "off" button. Maybe I just had a typo in the command sequence or a bug in the code when I tested it, who knows.

By the way, the light and the remote have channel controls – I assume that either the prefix or suffix in the command protocol somehow encodes which of the eight channels the command is meant for.
